Area chart changes (#209)
patternfly/training-scenarios: react-charts/area-chart/step4.md

Add tooltips to the area chart.

1) <strong>Make sure the `App.js` file is still open.</strong>

2) <strong>Locate the `<Chart>` component.</strong>

It will look like this:

<pre class="file">
&lt;Chart
  height={250}
  width={600}
&gt;
</pre>

> 3) <strong>Add a `containerComponent` property to that section</strong>

Reviewer: Should we replace 'that section' with 'the <ComponentName> component' in all steps?
```suggestion
3) <strong>Add a `containerComponent` property to the `Chart` component</strong>
```

The `containerComponent` property takes a component instance, which will be used to create a container element for standalone charts.

Add the `ChartVoronoiContainer` component which adds the ability to associate a mouse position with the data point(s) closest to it. This is useful for adding a tooltip.

It should look like this:

<pre class="file" data-target="clipboard">
containerComponent={
  &lt;ChartVoronoiContainer
  /&gt;
}
</pre>

<strong>Note:</strong> There are no visible changes at this stage.

4) <strong>Add a `constrainToVisibleArea` property inside of the `ChartVoronoiContainer`.</strong>

> The `constrainToVisibleArea` property will alter the position of the tooltip so that it exactly fits within the rendered SVG.

Reviewer: We are using a little bit of jargon here.
Victory has a pretty good description in their docs:
```suggestion
The `constrainToVisibleArea` property determines whether to coerce tooltips so that they fit within the visible area of the chart.
```

In this step you will add tooltips to the area chart.

1) <strong>Make sure the `App.js` file is still open.</strong>

2) <strong>Locate the `<Chart>` component.</strong>

It will look like this:

<pre class="file">
&lt;Chart
  height={250}
  width={600}
&gt;
</pre>

3) <strong>Add a `containerComponent` property to that section</strong>

The `containerComponent` property takes a component instance which will be used to create a container element for standalone charts.

> You will add the `ChartVoronoiContainer` component which adds the ability to associate a mouse position with the data point(s) closest to it. This is useful for adding a tooltip.

Reviewer 2: I need some clarification. During the intro, the future tense would be acceptable, correct? And present tense within each step?

> You will add the `ChartVoronoiContainer` component which adds the ability to associate a mouse position with the data point(s) closest to it. This is useful for adding a tooltip.

Reviewer 3: Future tense should be okay with the intros since we are telling them something they'll do in the future! Sorry if some present tense suggestions snuck in for intros!

In this step you will add tooltips to the area chart.

1) <strong>Make sure the `App.js` file is still open.</strong>

2) <strong>Locate the `<Chart>` component.</strong>

It will look like this:

<pre class="file">
&lt;Chart
  height={250}
  width={600}
&gt;
</pre>

> 3) <strong>Add a `containerComponent` property to that section:</strong>

Reviewer 4: Adding `containerComponent={}` breaks the example, which is why I did not split this up into multiple steps. IMO, this does not add value as a separate step.

Author: I have combined the `containerComponent={}` and the <ChartVoronoiContainer/> in one step so that it still breaks up the steps, but there is no error.

We just got a lot of feedback that users wanted the steps to be broken up more, but I 100% agree that steps that throw errors shouldn't be split up.

The `containerComponent` property takes a component instance which will be used to create a container element for standalone charts.

It should look like this:

<pre class="file" data-target="clipboard">
containerComponent={}
</pre>

> 4) <strong>Add a `ChartVoronoiContainer` inside of the `containerComponent`.</strong>

Reviewer 4: Adding ChartVoronoiContainer does nothing without the labels. 

I believe the intent is for each step to demonstrate a unique feature. However, it all needs to work together in this case.

Author: Understood. There are quite a few steps with the purpose of just setting up a structure and they don't show any visible changes in the UI, so this would be one of those cases. It allows the user to understand how/why the component works vs seeing the final result.

Reviewer 4: As long as there is no breakage.

> The `ChartVoronoiContainer` is used to create voronoi overlays for charts, which are useful for attaching events to data points, such as tooltips.

Reviewer 5: What are these *voronoi overlays*? I would at least add link to the https://formidable.com/open-source/victory/docs/victory-voronoi-container/

Reviewer 4: Perhaps replace "voronoi overlays" with...

"VictoryVoronoiContainer adds the ability to associate a mouse position with the data point(s) closest to it"

Author: @dlabrecq thanks for that description, I'll update

It should look like this:

<pre class="file" data-target="clipboard">
containerComponent={
  &lt;ChartVoronoiContainer
  /&gt;
}
</pre>

5) <strong>Add a `constrainToVisibleArea` property inside of the `ChartVoronoiContainer`.</strong>

> 

Reviewer 4: I'm not seeing any tooltips after completing these steps.

Author: Captured in this issue: https://github.com/patternfly/training-scenarios/issues/210
